### PR TITLE
Add "Gradient Descent" to AI topic

### LINKED ISSUE #2018: [Concept Entry] AI Gradient Descent
### Reason/inspiration (optional)

We would like a new concept entry for Gradient Ascent

The entry should go in a new file in docs/content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md

The entry should
- Describe the method of Gradient Descent
- Describe its relation to AI.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by YanisaHS**
### Description

I added "Gradient Descent" in the AI topic under "search algorithms". I think it would be better suited under an "optimization algorithm" section though? Not sure if that's planned, but just mentioning.

Closes #2018 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+Title: 'Gradient Descent'
+Description: 'Gradient descent is an optimization search algorithm that is widely used in machine learning to train neural networks and other models.'
+Subjects:
+  - 'Data Science'
+  - 'AI'
+Tags:
+  - 'Search Algorithms'
```
Comment (caupolicandiaz) [inline content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md:8]:
```suggestion
  - 'Search'
  - 'Algorithms'
```

File: content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## How Does Gradient Descent Work?
+
+Gradient descent works by minimizing the cost (or loss) function of a model. Cost functions measure the performance of machine learning models by quantifying the difference between the predicted and actual outputs. A low cost function indicates that a model is performing well on the training data because its predicted output values are close to the actual output values.
+
+Gradient descent is an iterative optimization algorithm, which means that it iteratively, or repeatedly, updates the parameters of a model until a specific outcome is achieved. Parameters can be thought of as the “settings” for a machine learning model. The purpose of gradient descent is to iterate until the local or global minimum is found in a given space or “local neighborhood”. The “gradient” refers to the rate of change, or slope, of the cost function. Thus, a gradient descent algorithm iteratively moves down (descends) the gradient (slope of the cost function) to find the minimum point, which represents the optimal values for the model to achieve good performance.
```
Comment (caupolicandiaz) [inline content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md:21]:
```suggestion
Gradient descent is an iterative optimization algorithm, which means that it iteratively, or repeatedly, updates the parameters of a model until a specific outcome is achieved. Parameters can be thought of as the “settings” for a machine learning model. The purpose of gradient descent is to iterate until the local or global minimum is found in a given solution space or “local neighborhood”. The “gradient” refers to the rate of change, or slope, of the cost function. Thus, a gradient descent algorithm iteratively moves down (descends) the gradient (slope of the cost function) to find the minimum point, which represents the optimal values (the best performance) for the model in the given context.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@YanisaHS Overall, this looks really good. I've added just a couple of comments. Please update the entry, and tag me when you've finished making the edits. Thanks

**Comment by YanisaHS** [comment]
@caupolicandiaz thanks for your review! Done 🥳

**Review by caupolicandiaz — APPROVED** [review]
@YanisaHS This is ready for another review.

**Review by HishamT — APPROVED** [review]
Very well written! LGTM! :grin:

**Comment by YanisaHS** [comment]
@HishamT thanks for your review! Would you mind looking at the change I just added? I noticed a typo 🥳 Sorry if the change looks weird on your end though? I was exploring GitHub mobile app 👀

File: content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## How Does Gradient Descent Work?
+
+Gradient descent works by minimizing the cost (or loss) function of a model. Cost functions measure the performance of machine learning models by quantifying the difference between the predicted and actual outputs. A low cost function indicates that a model is performing well on the training data because its predicted output values are close to the actual output values.
+
+Gradient descent is an iterative optimization algorithm, which means that it iteratively, or repeatedly, updates the parameters of a model until a specific outcome is achieved. Parameters can be thought of as the “settings” for a machine learning model. The purpose of gradient descent is to iterate until the local or global minimum is found in a given solution space or “local neighborhood”. The “gradient” refers to the rate of change, or slope, of the cost function. Thus, a gradient descent algorithm iteratively moves down (descends) the gradient (slope of the cost function) to find the minimum point, which represents the optimal values (the best performance) for the model in the given context.
```
Comment (yangc95) [inline content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md:22]:
```suggestion
Gradient descent is an iterative optimization algorithm, which means that it iteratively, or repeatedly, updates the parameters of a model until a specific outcome is achieved. Parameters can be thought of as the “settings” for a machine learning model. The purpose of gradient descent is to iterate until the local or global minimum is found in a given solution space or “local neighborhood." The “gradient” refers to the rate of change, or slope, of the cost function. Thus, a gradient descent algorithm iteratively moves down (descends) the gradient (slope of the cost function) to find the minimum point, which represents the optimal values (the best performance) for the model in the given context.
```

Comment (yangc95) [reply content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md:22]: very minor suggestion

**Review by HishamT — APPROVED** [review]
@YanisaHS, still good! :grin:

**Comment by yangc95** [comment]
🥳 @YanisaHS live link: https://www.codecademy.com/resources/docs/ai/search-algorithms/gradient-descent

Screenshot 2023-05-18 at 6 50 36 PM

cc: @caupolicandiaz @HishamT

**Pull request merged**